test/test_torch.py

```diff
@@ -7022,6 +7022,44 @@ def assert_with_filename(filename):
             with TemporaryDirectoryName(suffix='中文') as dname, TemporaryFileName(dir=dname) as fname:
                 assert_with_filename(fname)
 
+    def test_from_file_offset(self):
+        def assert_with_filename(filename):
```

Comment:
nit: remove the function and just use the context manager for the file? Or do you plan to add more here and you want to keep this structure?